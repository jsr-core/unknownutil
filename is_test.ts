import { assertEquals } from "https://deno.land/std@0.186.0/testing/asserts.ts";
import {
  isArray,
  isBoolean,
  isFunction,
  isNull,
  isNullish,
  isNumber,
  isObject,
  isString,
  isUndefined,
} from "./is.ts";

Deno.test("isString returns true on array", () => {
  assertEquals(isString(""), true);
  assertEquals(isString("Hello World"), true);
});
Deno.test("isString returns false on non array", () => {
  assertEquals(isString(0), false);
  assertEquals(isString(true), false);
  assertEquals(isString(false), false);
  assertEquals(isString([]), false);
  assertEquals(isString({}), false);
  assertEquals(isString(function () {}), false);
  assertEquals(isString(null), false);
  assertEquals(isString(undefined), false);
});

Deno.test("isNumber returns true on array", () => {
  assertEquals(isNumber(0), true);
  assertEquals(isNumber(1234567890), true);
});
Deno.test("isNumber returns false on non array", () => {
  assertEquals(isNumber(""), false);
  assertEquals(isNumber(true), false);
  assertEquals(isNumber(false), false);
  assertEquals(isNumber([]), false);
  assertEquals(isNumber({}), false);
  assertEquals(isNumber(function () {}), false);
  assertEquals(isNumber(null), false);
  assertEquals(isNumber(undefined), false);
});

Deno.test("isBoolean returns true on boolean", () => {
  assertEquals(isBoolean(true), true);
  assertEquals(isBoolean(false), true);
});
Deno.test("isBoolean returns false on non boolean", () => {
  assertEquals(isBoolean(0), false);
  assertEquals(isBoolean(""), false);
  assertEquals(isBoolean([]), false);
  assertEquals(isBoolean({}), false);
  assertEquals(isBoolean(function () {}), false);
  assertEquals(isBoolean(null), false);
  assertEquals(isBoolean(undefined), false);
});

Deno.test("isArray returns true on array", () => {
  assertEquals(isArray([]), true);
  assertEquals(isArray([0, 1, 2]), true);
  assertEquals(isArray(["a", "b", "c"]), true);
  assertEquals(isArray([0, "a", 1]), true);
});
Deno.test("isArray returns false on non array", () => {
  assertEquals(isArray(""), false);
  assertEquals(isArray(0), false);
  assertEquals(isArray(true), false);
  assertEquals(isArray(false), false);
  assertEquals(isArray({}), false);
  assertEquals(isArray(function () {}), false);
  assertEquals(isArray(null), false);
  assertEquals(isArray(undefined), false);
});
Deno.test("isArray<T> returns true on T array", () => {
  assertEquals(isArray([0, 1, 2], isNumber), true);
  assertEquals(isArray(["a", "b", "c"], isString), true);
  assertEquals(isArray([true, false, true], isBoolean), true);
});
Deno.test("isArray<T> returns false on non T array", () => {
  assertEquals(isArray([0, 1, 2], isString), false);
  assertEquals(isArray(["a", "b", "c"], isNumber), false);
  assertEquals(isArray([true, false, true], isString), false);
});

Deno.test("isObject returns true on object", () => {
  assertEquals(isObject({}), true);
  assertEquals(isObject({ a: 0 }), true);
  assertEquals(isObject({ a: "a" }), true);
});
Deno.test("isObject returns false on non object", () => {
  assertEquals(isObject(""), false);
  assertEquals(isObject(0), false);
  assertEquals(isObject(true), false);
  assertEquals(isObject(false), false);
  assertEquals(isObject([]), false);
  assertEquals(isObject(function () {}), false);
  assertEquals(isObject(null), false);
  assertEquals(isObject(undefined), false);
});
Deno.test("isObject<T> returns true on T object", () => {
  assertEquals(isObject({ a: 0 }, isNumber), true);
  assertEquals(isObject({ a: "a" }, isString), true);
  assertEquals(isObject({ a: true }, isBoolean), true);
});
Deno.test("isObject<T> returns false on non T object", () => {
  assertEquals(isObject({ a: 0 }, isString), false);
  assertEquals(isObject({ a: "a" }, isNumber), false);
  assertEquals(isObject({ a: true }, isString), false);
});

Deno.test("isFunction returns true on function", () => {
  assertEquals(isFunction(isFunction), true);
  assertEquals(isFunction(function () {}), true);
  assertEquals(isFunction(() => {}), true);
  assertEquals(isFunction(setTimeout), true);
});
Deno.test("isFunction returns false on non function", () => {
  assertEquals(isFunction(""), false);
  assertEquals(isFunction(0), false);
  assertEquals(isFunction(true), false);
  assertEquals(isFunction(false), false);
  assertEquals(isFunction([]), false);
  assertEquals(isFunction({}), false);
  assertEquals(isFunction(null), false);
  assertEquals(isFunction(undefined), false);
});

Deno.test("isNull returns true on null", () => {
  assertEquals(isNull(null), true);
});
Deno.test("isNull returns false on non null", () => {
  assertEquals(isNull(""), false);
  assertEquals(isNull(0), false);
  assertEquals(isNull(true), false);
  assertEquals(isNull(false), false);
  assertEquals(isNull([]), false);
  assertEquals(isNull({}), false);
  assertEquals(isNull(function () {}), false);
  assertEquals(isNull(undefined), false);
});

Deno.test("isUndefined returns true on null", () => {
  assertEquals(isUndefined(undefined), true);
});
Deno.test("isUndefined returns false on non null", () => {
  assertEquals(isUndefined(""), false);
  assertEquals(isUndefined(0), false);
  assertEquals(isUndefined(true), false);
  assertEquals(isUndefined(false), false);
  assertEquals(isUndefined([]), false);
  assertEquals(isUndefined({}), false);
  assertEquals(isUndefined(function () {}), false);
  assertEquals(isUndefined(null), false);
});

Deno.test("isNullish returns true on null/undefined", () => {
  assertEquals(isNullish(null), true);
  assertEquals(isNullish(undefined), true);
});
Deno.test("isNullish returns false on non null/undefined", () => {
  assertEquals(isNullish(""), false);
  assertEquals(isNullish(0), false);
  assertEquals(isNullish(true), false);
  assertEquals(isNullish(false), false);
  assertEquals(isNullish([]), false);
  assertEquals(isNullish({}), false);
  assertEquals(isNullish(function () {}), false);
});
