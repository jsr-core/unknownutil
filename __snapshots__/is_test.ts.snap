export const snapshot = {};

snapshot[`is.OptionalOf<T> > returns properly named function 1`] = `"isOptionalOf(isNumber)"`;

snapshot[`is.OptionalOf<T> > returns properly named function 2`] = `"isOptionalOf(isNumber)"`;

snapshot[`is.UnwrapOptionalOf<T> > returns properly named function 1`] = `"isNumber"`;

snapshot[`is.UnwrapOptionalOf<T> > returns properly named function 2`] = `"isNumber"`;

snapshot[`is.UnwrapOptionalOf<T> > returns properly named function 3`] = `"isNumber"`;

snapshot[`is.ReadonlyOf<T> > returns properly named function 1`] = `"isReadonlyOf(isNumber)"`;

snapshot[`is.ReadonlyOf<T> > returns properly named function 2`] = `"isReadonlyOf(isReadonlyOf(isNumber))"`;

snapshot[`is.UnwrapReadonlyOf<T> > returns properly named function 1`] = `"isNumber"`;

snapshot[`is.UnwrapReadonlyOf<T> > returns properly named function 2`] = `"isReadonlyOf(isNumber)"`;

snapshot[`is.ArrayOf<T> > returns properly named function 1`] = `"isArrayOf(isNumber)"`;

snapshot[`is.ArrayOf<T> > returns properly named function 2`] = `"isArrayOf((anonymous))"`;

snapshot[`is.SetOf<T> > returns properly named function 1`] = `"isSetOf(isNumber)"`;

snapshot[`is.SetOf<T> > returns properly named function 2`] = `"isSetOf((anonymous))"`;

snapshot[`is.TupleOf<T> > returns properly named function 1`] = `
"isTupleOf([
  isNumber,
  isString,
  isBoolean
])"
`;

snapshot[`is.TupleOf<T> > returns properly named function 2`] = `"isTupleOf([(anonymous)])"`;

snapshot[`is.TupleOf<T> > returns properly named function 3`] = `
"isTupleOf([
  isTupleOf([
    isTupleOf([
      isNumber,
      isString,
      isBoolean
    ])
  ])
])"
`;

snapshot[`is.TupleOf<T, E> > returns properly named function 1`] = `
"isTupleOf([
  isNumber,
  isString,
  isBoolean
], isArray)"
`;

snapshot[`is.TupleOf<T, E> > returns properly named function 2`] = `"isTupleOf([(anonymous)], isArrayOf(isString))"`;

snapshot[`is.TupleOf<T, E> > returns properly named function 3`] = `
"isTupleOf([
  isTupleOf([
    isTupleOf([
      isNumber,
      isString,
      isBoolean
    ], isArray)
  ], isArray)
])"
`;

snapshot[`is.ParametersOf<T> > returns properly named function 1`] = `
"isParametersOf([
  isNumber,
  isString,
  isOptionalOf(isBoolean)
])"
`;

snapshot[`is.ParametersOf<T> > returns properly named function 2`] = `"isParametersOf([(anonymous)])"`;

snapshot[`is.ParametersOf<T> > returns properly named function 3`] = `"isParametersOf([])"`;

snapshot[`is.ParametersOf<T> > returns properly named function 4`] = `
"isParametersOf([
  isParametersOf([
    isParametersOf([
      isNumber,
      isString,
      isOptionalOf(isBoolean)
    ])
  ])
])"
`;

snapshot[`is.ParametersOf<T, E> > returns properly named function 1`] = `
"isParametersOf([
  isNumber,
  isString,
  isOptionalOf(isBoolean)
], isArray)"
`;

snapshot[`is.ParametersOf<T, E> > returns properly named function 2`] = `"isParametersOf([(anonymous)], isArrayOf(isString))"`;

snapshot[`is.ParametersOf<T, E> > returns properly named function 3`] = `"isParametersOf([], isArrayOf(isString))"`;

snapshot[`is.ParametersOf<T, E> > returns properly named function 4`] = `
"isParametersOf([
  isParametersOf([
    isParametersOf([
      isNumber,
      isString,
      isOptionalOf(isBoolean)
    ], isArray)
  ], isArray)
])"
`;

snapshot[`is.UniformTupleOf<T> > returns properly named function 1`] = `"isUniformTupleOf(3, isAny)"`;

snapshot[`is.UniformTupleOf<T> > returns properly named function 2`] = `"isUniformTupleOf(3, isNumber)"`;

snapshot[`is.UniformTupleOf<T> > returns properly named function 3`] = `"isUniformTupleOf(3, (anonymous))"`;

snapshot[`is.RecordObjectOf<T> > returns properly named function 1`] = `"isRecordObjectOf(isNumber, undefined)"`;

snapshot[`is.RecordObjectOf<T> > returns properly named function 2`] = `"isRecordObjectOf((anonymous), undefined)"`;

snapshot[`is.RecordObjectOf<T, K> > returns properly named function 1`] = `"isRecordObjectOf(isNumber, isString)"`;

snapshot[`is.RecordObjectOf<T, K> > returns properly named function 2`] = `"isRecordObjectOf((anonymous), isString)"`;

snapshot[`is.RecordOf<T> > returns properly named function 1`] = `"isRecordOf(isNumber, undefined)"`;

snapshot[`is.RecordOf<T> > returns properly named function 2`] = `"isRecordOf((anonymous), undefined)"`;

snapshot[`is.RecordOf<T, K> > returns properly named function 1`] = `"isRecordOf(isNumber, isString)"`;

snapshot[`is.RecordOf<T, K> > returns properly named function 2`] = `"isRecordOf((anonymous), isString)"`;

snapshot[`is.MapOf<T> > returns properly named function 1`] = `"isMapOf(isNumber, undefined)"`;

snapshot[`is.MapOf<T> > returns properly named function 2`] = `"isMapOf((anonymous), undefined)"`;

snapshot[`is.MapOf<T, K> > returns properly named function 1`] = `"isMapOf(isNumber, isString)"`;

snapshot[`is.MapOf<T, K> > returns properly named function 2`] = `"isMapOf((anonymous), isString)"`;

snapshot[`is.ObjectOf<T> > returns properly named function 1`] = `
"isObjectOf({
  a: isNumber,
  b: isString,
  c: isBoolean
})"
`;

snapshot[`is.ObjectOf<T> > returns properly named function 2`] = `"isObjectOf({a: a})"`;

snapshot[`is.ObjectOf<T> > returns properly named function 3`] = `
"isObjectOf({
  a: isObjectOf({
    b: isObjectOf({c: isBoolean})
  })
})"
`;

snapshot[`is.StrictOf<T> > returns properly named function 1`] = `
"isStrictOf(isObjectOf({
  a: isNumber,
  b: isString,
  c: isBoolean
}))"
`;

snapshot[`is.StrictOf<T> > returns properly named function 2`] = `"isStrictOf(isObjectOf({a: a}))"`;

snapshot[`is.StrictOf<T> > returns properly named function 3`] = `
"isStrictOf(isObjectOf({
  a: isStrictOf(isObjectOf({
    b: isStrictOf(isObjectOf({c: isBoolean}))
  }))
}))"
`;

snapshot[`is.InstanceOf<T> > returns properly named function 1`] = `"isInstanceOf(Date)"`;

snapshot[`is.InstanceOf<T> > returns properly named function 2`] = `"isInstanceOf((anonymous))"`;

snapshot[`is.LiteralOf<T> > returns properly named function 1`] = `'isLiteralOf("hello")'`;

snapshot[`is.LiteralOf<T> > returns properly named function 2`] = `"isLiteralOf(100)"`;

snapshot[`is.LiteralOf<T> > returns properly named function 3`] = `"isLiteralOf(100n)"`;

snapshot[`is.LiteralOf<T> > returns properly named function 4`] = `"isLiteralOf(true)"`;

snapshot[`is.LiteralOf<T> > returns properly named function 5`] = `"isLiteralOf(null)"`;

snapshot[`is.LiteralOf<T> > returns properly named function 6`] = `"isLiteralOf(undefined)"`;

snapshot[`is.LiteralOf<T> > returns properly named function 7`] = `"isLiteralOf(Symbol(asdf))"`;

snapshot[`is.LiteralOneOf<T> > returns properly named function 1`] = `'isLiteralOneOf(["hello", "world"])'`;

snapshot[`is.UnionOf<T> > returns properly named function 1`] = `
"isUnionOf([
  isNumber,
  isString,
  isBoolean
])"
`;

snapshot[`is.IntersectionOf<T> > returns properly named function 1`] = `
"isObjectOf({
  a: isNumber,
  b: isString
})"
`;

snapshot[`is.IntersectionOf<T> > returns properly named function 2`] = `"isString"`;

snapshot[`is.IntersectionOf<T> > returns properly named function 3`] = `
"isIntersectionOf([
  isFunction,
  isObjectOf({b: isString})
])"
`;

snapshot[`is.RequiredOf<T> > returns properly named function 1`] = `
"isObjectOf({
  a: isNumber,
  b: isUnionOf([
    isString,
    isUndefined
  ]),
  c: isBoolean
})"
`;

snapshot[`is.RequiredOf<T> > returns properly named function 2`] = `
"isObjectOf({
  a: isNumber,
  b: isUnionOf([
    isString,
    isUndefined
  ]),
  c: isBoolean
})"
`;

snapshot[`is.PartialOf<T> > returns properly named function 1`] = `
"isObjectOf({
  a: isOptionalOf(isNumber),
  b: isOptionalOf(isUnionOf([
    isString,
    isUndefined
  ])),
  c: isOptionalOf(isBoolean)
})"
`;

snapshot[`is.PartialOf<T> > returns properly named function 2`] = `
"isObjectOf({
  a: isOptionalOf(isNumber),
  b: isOptionalOf(isUnionOf([
    isString,
    isUndefined
  ])),
  c: isOptionalOf(isBoolean)
})"
`;

snapshot[`is.PickOf<T, K> > returns properly named function 1`] = `
"isObjectOf({
  a: isNumber,
  c: isBoolean
})"
`;

snapshot[`is.PickOf<T, K> > returns properly named function 2`] = `"isObjectOf({a: isNumber})"`;

snapshot[`is.OmitOf<T, K> > returns properly named function 1`] = `
"isObjectOf({
  a: isNumber,
  c: isBoolean
})"
`;

snapshot[`is.OmitOf<T, K> > returns properly named function 2`] = `"isObjectOf({a: isNumber})"`;
